xde/src/xde.rs

```diff
@@ -1921,34 +2694,39 @@ unsafe extern "C" fn xde_mc_tx(
     let mut hairpin_chain = MsgBlkChain::empty();
     let mut tx_postbox = TxPostbox::new();
 
-    // We don't need to read-lock the port map unless we have local
-    // delivery to perform.
-    //
-    // TODO: really think this one through. This might expose us to the
-    // risk of double read-locking at the same time as the tokenlock
-    // wants to make some globally mutable operation happen.
-    //
-    // Maybe we should clone out the `DevMap` at this instant.
-    let mut entry_state = None;
+    // Clone per-CPU mcast forwarding table Arc and drop lock immediately.
+    // This makes the reader lock-free and avoids blocking management refreshes.
+    let cpu_index = current_cpu().seq_id;
+    let cpu_entry = &src_dev.u1.stream.ports_map[cpu_index];
+    let mcast_fwd = clone_from_rwlock(&cpu_entry.mcast_fwd);
+
+    // Lazily clone per-port DevMap Arc for hairpin/local delivery.
```

Comment:
@FelixMcFelix, of note, this is a different approach I took with Arc cloning `mcast_fwd` and then dropping the lock. Take a gander to see if you have worries here or below.